pkg/operation/botanist/component/metricsserver/metrics_server.go

```diff
@@ -134,6 +136,21 @@ func (m *metricsServer) computeResourcesData(serverSecret, caSecret *corev1.Secr
 	}
 	utilruntime.Must(kutil.MakeUnique(secret))
 
+	// Required RBAC resources change in v0.6.0. Ref: https://github.com/kubernetes-sigs/metrics-server/releases/tag/v0.6.0
+	if m.image == nil || m.image.Tag == nil {
+		return nil, fmt.Errorf("image and image.tag must not be nil")
+	}
+	// v0.6.0 rbac resources
+	rbacResources := []string{"pods", "nodes", "nodes/metrics", "namespaces", "configmaps"}
+	preVersionZeroSix, err := version.CheckVersionMeetsConstraint(*m.image.Tag, "< v0.6.0")
```

Comment:
I don't think it is fine to rely on the tag. You know that we have a concept/feature about image vector overwrite - see [`./docs/deployment/image_vector.md`](https://github.com/gardener/gardener/blob/v1.51.0/docs/deployment/image_vector.md). We cannot rely that the tag will be preserved as v0.6.0 if image vector overwrite is used. I think even sometimes our CI/CD overwrites the tags with some suffixes - for example `v0.6.0` becomes `v0.6.0-mod1` (don't ask me why, I don't know). 
I would rather suggest to use the K8s version of the Shoot. For K8s < 1.19 use the old RBAC, for K8s >= 1.19 use the new one.

Reply:
In the recent nginx update PR, I have seen that we don't necessary create different cases for RBAC changes https://github.com/gardener/gardener/pull/6356/files#r924183792
I could do the same here and add both resources and we would have only minimal changes here. WDYT?